tests/unit/workflows/test_workflows_tasks_texkeys.py

```diff
@@ -0,0 +1,435 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+from __future__ import absolute_import, division, print_function
+
+import mock
+import pytest
+
+from inspire_dojson.utils import validate
+from inspire_schemas.utils import load_schema
+from inspirehep.modules.records.texkeys import inspire_texkey_minter
+
+
+@mock.patch(
+    'inspirehep.modules.records.texkeys.get_three_random_letters',
+    return_value='xyz',
+)
+@mock.patch(
+    'inspirehep.modules.records.texkeys.store_texkeys_pidstore',
+    return_value=True,
+)
+def test_inspire_texkey_minter_with_author(mock_store_pidstore, mock_random_letters):
+    record = {
+        'authors': [
+            {
+                'inspire_role': 'author',
+                'full_name': 'Cox, Brian',
+            }, {
+                'inspire_role': 'author',
+                'full_name': 'Dan, Brown'
+            }
+        ]
+    }
+    inspire_texkey_minter('', record)
+
+    expected = {
+        'texkeys': ['Cox:2017xyz'],
+        'authors': [
+            {
+                'inspire_role': 'author',
+                'full_name': 'Cox, Brian',
+            }, {
+                'inspire_role': 'author',
+                'full_name': 'Dan, Brown'
+            }
+        ]
+    }
+
+    assert record == expected
+
+    schema = load_schema('hep')
+    subschema_texkeys = schema['properties']['texkeys']
+
+    assert validate(record['texkeys'], subschema_texkeys) is None
+
+
+@mock.patch(
+    'inspirehep.modules.records.texkeys.get_three_random_letters',
+    return_value='xyz',
+)
+@mock.patch(
+    'inspirehep.modules.records.texkeys.store_texkeys_pidstore',
+    return_value=True,
+)
+def test_inspire_texkey_minter_with_more_than_10_authors(mock_store_pidstore, mock_random_letters):
+    record = {
+        'authors': [
+            {
+                'full_name': 'Arnaldi, R.'
+            }, {
+                'full_name': 'Alexa, C.'
+            }, {
+                'full_name': 'Alessandro, B.'
+            }, {
+                'full_name': 'Atayan, M.'
+            }, {
+                'full_name': 'Beole, S.'
+            }, {
+                'full_name': 'Boldea, V.'
+            }, {
+                'full_name': 'Bordalo, P.'
+            }, {
+                'full_name': 'Borges, G.'
+            }, {
+                'full_name': 'Castanier, C.'
+            }, {
+                'full_name': 'Castor, J.'
+            }, {
+                'full_name': 'Chaurand, B.'
+            }
```

Comment:
Even thought in everyday life I prefer this style of indentation, in this case it's definitely better use the compact one!